pages/constructor-encoder.page.tsx

```diff
@@ -0,0 +1,171 @@
+import { Interface } from '@ethersproject/abi';
+import { BigNumber } from '@ethersproject/bignumber';
+import { isValidAddress, stripHexPrefix } from 'ethereumjs-util';
+import _ from 'lodash';
+import { ChangeEvent, useState } from 'react';
+
+import DecoderSvg from '../public/static/svg/decoders';
+import { Button } from '../src/components/lib/Button';
+import { Input } from '../src/components/lib/Input';
+import { ToolLayout } from '../src/layout/ToolLayout';
+import { encodeConstructor } from '../src/lib/encodeContructor';
+import { parseAbi } from '../src/lib/parseAbi';
+
+export interface ConstructorEncoderProps {}
+
+export default function ConstructorEncoder() {
+  const [error, setError] = useState<string | undefined>();
+  const [encodedResult, setEncodedResult] = useState<string[] | undefined>();
+
+  const [rawAbi, setRawAbi] = useState<string>();
+  const [iface, setIface] = useState<Interface>();
+  const [errors, setErrors] = useState<string[]>([]);
+  const [values, setValues] = useState<string[]>([]);
+
+  async function handleEncodeConstructor() {
+    if (!rawAbi) {
+      setError('Please provide ABI');
+      return;
+    }
+    if (_.some(values, _.isUndefined)) {
```

Comment:
1. Can this actually occur? `values` has a type `string[]`. Is the type wrong or is this condition a leftover from a previous version of the code?

2. Wouldn't we rather write `values.some(isUndefined)`? `Array.prototype.some` was specified in ES5 and has been widely supported for a longer while.